aeppl/transforms.py

```diff
@@ -47,50 +51,52 @@ def class_transformed_rv(op, node):
         return cls_res
 
 
-class RVTransform(abc.ABC):
-    @abc.abstractmethod
-    def forward(self, value: TensorVariable, *inputs: Variable) -> TensorVariable:
-        """Apply the transformation."""
+class TransformedRV(RandomVariable, metaclass=TransformedRVMeta):
+    r"""A base class for transformed `RandomVariable`\s."""
 
-    @abc.abstractmethod
-    def backward(self, value: TensorVariable, *inputs: Variable) -> TensorVariable:
-        """Invert the transformation."""
 
-    def log_jac_det(self, value: TensorVariable, *inputs) -> TensorVariable:
-        """Construct the log of the absolute value of the Jacobian determinant."""
-        # jac = at.reshape(
-        #     gradient(at.sum(self.backward(value, *inputs)), [value]), value.shape
-        # )
-        # return at.log(at.abs_(jac))
-        phi_inv = self.backward(value, *inputs)
-        return at.log(at.nlinalg.det(at.atleast_2d(jacobian(phi_inv, [value]))))
+@singledispatch
```

Comment:
This was moved here, so that in the next commit I can change the return type to `Optional[TransformedRV]` without additional changes